cloudinit/sources/DataSourceVMware.py

```diff
@@ -131,14 +173,33 @@ class DataSourceVMware(sources.DataSource):
     def __init__(self, sys_cfg, distro, paths, ud_proc=None):
         sources.DataSource.__init__(self, sys_cfg, distro, paths, ud_proc)
 
+        self.cfg = {}
         self.data_access_method = None
         self.vmware_rpctool = VMWARE_RPCTOOL
 
+        # A list includes all possible data transports, each tuple represents
+        # one data transport type. This datasource will try to get data from
+        # each of transports follows the tuples order in this list.
+        # A tuple has 3 elements which are:
+        # 1. The transport name
+        # 2. The function name to get data for the transport
+        # 3. A boolean tells whether the transport requires VMware platform
+        self.possible_data_access_method_list = [
+            (DATA_ACCESS_METHOD_VMWARE_IMC, self.get_vmware_imc_data_fn, True),
+            (DATA_ACCESS_METHOD_ENVVAR, self.get_envvar_data_fn, False),
+            (DATA_ACCESS_METHOD_GUESTINFO, self.get_guestinfo_data_fn, True),
+        ]
+
+    def __str__(self):
+        root = sources.DataSource.__str__(self)
+        return "%s [seed=%s]" % (root, self.data_access_method)
+
     def _get_data(self):
         """
         _get_data loads the metadata, userdata, and vendordata from one of
         the following locations in the given order:
 
+            * vmware-imc
```

Comment:
I will not keep repeating the same comment (last one, I promise), but I think we should reconsider the order.

Reply:
ditto.